plugins/content/caption/base/lib/kMultiCaptionFlowManager.php

```diff
@@ -39,4 +39,49 @@ private function updatedParseMulti($dbBatchJob,$data)
 		$captionAsset->setStatus(asset::ASSET_STATUS_ERROR);
 		$captionAsset->save();
 	}
+	
+	/* (non-PHPdoc)
+ 	 * @see kObjectAddedEventConsumer::shouldConsumeAddedEvent()
+ 	 */
+	public function shouldConsumeAddedEvent(BaseObject $object)
+	{
+		if($object instanceof entry && $object->getReplacedEntryId() && $object->getIsTemporary())
+			return true;
+		
+		return false;
+	}
+	
+	/* (non-PHPdoc)
+ 	 * @see kObjectAddedEventConsumer::objectAdded()
+ 	 */
+	public function objectAdded(BaseObject $object, BatchJob $raisedJob = null)
+	{
+		if($object instanceof entry && $object->getReplacedEntryId() && $object->getIsTemporary())
+		{
+			$this->copyCaptionAssets($object);
+		}
+		
+		return true;
+	}
+	
+	protected function copyCaptionAssets(entry $entry)
+	{
+		$originalEntryId = $entry->getReplacedEntryId();
+		$originalEntry = entryPeer::retrieveByPK($originalEntryId);
+		if(!$originalEntry)
+		{
+			KalturaLog::debug("Original entry with id [$originalEntryId], not found");
+			return true;
+		}
+		
+		KalturaLog::debug("Original entry id $originalEntryId");
+		KalturaLog::debug("Replacing entry id [{$entry->getId()}]");
+		
+		$captions = assetPeer::retrieveByEntryId($originalEntryId, array(CaptionPlugin::getAssetTypeCoreValue(CaptionAssetType::CAPTION)));
+		foreach($captions as $caption)
+		{
+			/* @var asset $caption */
+			$caption->copyToEntry($entry->getId());
+		}
+	}
```

Comment:
Should return something here, or remove the rerun true if original entry not found